### ISSUE TITLE
ODS AMI build failed due to missing permission on user odsverify-dev

### ISSUE THREAD

**Issue by stitakis** [opendevstack/ods-core#1008]
**Describe the bug**
ODSBox AMI build is failing due to missing permission on user `odsverify-dev`.

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
 amazon-ebs: Allow serviceaccount 'jenkins' of odsverify-cd to admin the environment projects
16:23:31      amazon-ebs: + oc policy add-role-to-user admin system:serviceaccount:odsverify-cd:jenkins -n odsverify-dev
16:23:31      amazon-ebs: Warning: ServiceAccount 'jenkins' not found
16:23:31      amazon-ebs: role "admin" added: "system:serviceaccount:odsverify-cd:jenkins"
16:23:31      amazon-ebs: + oc policy add-role-to-user admin system:serviceaccount:odsverify-cd:jenkins -n odsverify-test
16:23:31      amazon-ebs: Warning: ServiceAccount 'jenkins' not found
16:23:31      amazon-ebs: role "admin" added: "system:serviceaccount:odsverify-cd:jenkins"
16:23:31      amazon-ebs: + echo 'Grant serviceaccount '\''jenkins'\'' role '\''edit'\'' in odsverify-cd'
16:23:31      amazon-ebs: Grant serviceaccount 'jenkins' role 'edit' in odsverify-cd
16:23:31      amazon-ebs: + oc policy add-role-to-user edit --serviceaccount jenkins -n odsverify-cd
16:23:31      amazon-ebs: Warning: ServiceAccount 'jenkins' not found
16:23:31      amazon-ebs: role "edit" added: "jenkins"
16:23:31      amazon-ebs: + echo 'Allow to pull odsverify-dev images from odsverify-test'
16:23:31      amazon-ebs: Allow to pull odsverify-dev images from odsverify-test
16:23:31      amazon-ebs: + oc policy add-role-to-group system:image-puller system:serviceaccounts:odsverify-test -n odsverify-dev
16:23:31      amazon-ebs: role "system:image-puller" added: "system:serviceaccounts:odsverify-test"
16:23:31      amazon-ebs: + echo 'Allow odsverify-dev and odsverify-test to pull odsverify-cd images'
16:23:31      amazon-ebs: Allow odsverify-dev and odsverify-test to pull odsverify-cd images
16:23:31      amazon-ebs: + oc policy add-role-to-group system:image-puller system:serviceaccounts:odsverify-test -n odsverify-cd
16:23:31      amazon-ebs: role "system:image-puller" added: "system:serviceaccounts:odsverify-test"
16:23:31      amazon-ebs: + oc policy add-role-to-group system:image-puller system:serviceaccounts:odsverify-dev -n odsverify-cd
16:23:31      amazon-ebs: role "system:image-puller" added: "system:serviceaccounts:odsverify-dev"
16:23:31      amazon-ebs: + echo 'Grant serviceaccount '\''default'\'' role '\''image-builder'\'' to import images from other cluster'
16:23:31      amazon-ebs: Grant serviceaccount 'default' role 'image-builder' to import images from other cluster
16:23:31      amazon-ebs: + oc policy add-role-to-user system:image-builder --serviceaccount default -n odsverify-dev
16:23:31      amazon-ebs: role "system:image-builder" added: "default"
16:23:31      amazon-ebs: + oc policy add-role-to-user system:image-builder --serviceaccount default -n odsverify-test
16:23:31      amazon-ebs: role "system:image-builder" added: "default"
16:23:31      amazon-ebs: + '[' -n openshift ']'
16:23:31      amazon-ebs: + echo 'Seeding admins (openshift) ...'
16:23:31      amazon-ebs: Seeding admins (openshift) ...
16:23:31      amazon-ebs: + for admin_user in '${PROJECT_ADMINS//,/ }'
16:23:31      amazon-ebs: + echo '- seeding admin: openshift'
16:23:31      amazon-ebs: - seeding admin: openshift
16:23:31      amazon-ebs: + oc policy add-role-to-user admin openshift -n odsverify-dev
16:23:31      amazon-ebs: role "admin" added: "openshift"
16:23:31      amazon-ebs: + oc policy add-role-to-user admin openshift -n odsverify-test
16:23:31      amazon-ebs: role "admin" added: "openshift"
16:23:31      amazon-ebs: + oc policy add-role-to-user admin openshift -n odsverify-cd
16:23:31      amazon-ebs: role "admin" added: "openshift"
16:23:31      amazon-ebs: + echo 'Assign the owner as a label to the OpenShift project'
16:23:31      amazon-ebs: Assign the owner as a label to the OpenShift project
16:23:31      amazon-ebs: ++ echo openshift
16:23:31      amazon-ebs: ++ cut -d , -f1
16:23:31      amazon-ebs: ++ sed -r s/@/_at_/g
16:23:31      amazon-ebs: + namespace_owner=openshift
16:23:31      amazon-ebs: + oc label namespace odsverify-dev opendevstack.org/project.owner=openshift
16:23:31      amazon-ebs: Error from server (Forbidden): namespaces "odsverify-dev" is forbidden: User "system:serviceaccount:ods:jenkins" cannot patch namespaces in the namespace "odsverify-dev": no RBAC policy matched
16:23:31      amazon-ebs: [Pipeline] }
16:23:31      amazon-ebs: [Pipeline] // dir
16:23:31      amazon-ebs: [Pipeline] }
16:23:31      amazon-ebs: [Pipeline] // stage
16:23:31      amazon-ebs: [Pipeline] }
16:23:31      amazon-ebs: [Pipeline] // node
16:23:31      amazon-ebs: [Pipeline] }
16:23:31      amazon-ebs: [Pipeline] // podTemplate
16:23:31      amazon-ebs: [Pipeline] End of Pipeline
16:23:31      amazon-ebs: ERROR: script returned exit code 1
16:23:31      amazon-ebs: Finished: FAILURE
```

**Comment by stitakis** [comment]
@felipecruz91 the build has not been running successfully the last week. Probably this is a regression issue originated by this change #946 Could you please double check?

**Comment by clemensutschig** [comment]
@felipecruz91  - does the (jenkins) service account need admin rights on the project, or something? I am pretty surprised that there is an issue, as the SA is creating the project?

**Comment by clemensutschig** [comment]
oops - this cannot work:
from https://docs.openshift.com/container-platform/3.6/admin_guide/managing_networking.html

`The cluster administrator role is required to add labels to namespaces.` why are we not using annotations?!

**Comment by stitakis** [comment]
@clemensutschig @felipecruz91 I have fix for this, we need to add to the ods service account `jenkins` the capability to patch namespaces. The way that I found to do this is by creating a new cluster role.
This can be done in this way
```
# Allow Jenkins serviceaccount to label namespaces
oc create clusterrole namespace-labeler --verb=patch --resource=namespace
oc adm policy add-cluster-role-to-user namespace-labeler system:serviceaccount:${NAMESPACE}:jenkins
```   
These lines need to be added to the script `setup-ods-project.sh`.
I'll have tested this solution in the branch `feature/ods-devenv`.
I'll create a PR for master branch

**Comment by stitakis** [comment]
@felipecruz91 we have reverted #946. This was necessary to fix the broken ODS AMI build. Please see a related discussion in the candidate Fix PR #1011 that didn't make it. Any questions please contact me. Happy to explain to you all the details.
